reconcile/queries.py

```diff
@@ -959,18 +962,32 @@ def get_quay_orgs():
         }
         policies
       }
+      {% endif %}
+
+      {% if saas_files %}
       owned_saas_files {
         name
       }
+      {% endif %}
+
+      {% if sendgrid %}
+      sendgrid_accounts {
+        path
+        name
+      }
+      {% endif %}
     }
   }
 }
 """
 
 
-def get_roles():
+def get_roles(aws=True, saas_files=True, sendgrid=False):
     gqlapi = gql.get_api()
-    return gqlapi.query(ROLES_QUERY)['users']
+    query = Template(ROLES_QUERY).render(aws=aws,
```

Comment:
Templated queries is the only "interesting" part of this PR.

The rationale behind this is in order to avoid retrieving schemas that aren't necessary to the integration. If you fetch the roles in order to reconcile sendgrid teams, you don't care about the aws groups. If you include it in the query, the schema needs to be declared in the integration schema. Basically, we don't want to trigger the sendgrid PR integration in the PR check if the PR only includes aws groups changes.

Reply:
:heart:
i'm going to adopt this from now on!